src/fparser/two/Fortran2003.py

```diff
@@ -9234,6 +9243,293 @@ def match(string):
+        upper case) if there is a match or None if there is not.
+        :rtype: (str) or NoneType
+
+        '''
+        return STRINGBase.match(list(Intrinsic_Name.generic_function_names) +
+                                list(Intrinsic_Name.specific_function_names),
+                                string)
+
+
+class Intrinsic_Function_Reference(CallBase):  # No explicit rule
+    '''Represents Fortran intrinsics.
+
+    function-reference is intrinsic-name ( [ actual-arg-spec-list ] )
+
+    '''
+    subclass_names = []
+    use_names = ['Intrinsic_Name', 'Actual_Arg_Spec_List']
+
+    @staticmethod
+    def match(string):
+        '''Match the string as an intrinsic function. Also check that the
+        number of arguments provided matches the number expected by
+        the intrinsic.
+
+        :param str string: the string to match with the pattern rule.
+
+        :return: a tuple of size 2 containing the name of the \
+        intrinsic and its arguments if there is a match, or None if \
+        there is not.
+        :rtype: (:py:class:`fparser.two.Fortran2003.Intrinsic_Name`, \
+        :py:class:`fparser.two.Fortran2003.Actual_Arg_Spec_List`) or \
+        NoneType
+
+        :raises InternalSyntaxError: If the number of arguments \
+        provided does not match the number of arguments expected by \
+        the intrinsic.
+
+        '''
+        result = CallBase.match(
+            Intrinsic_Name, Actual_Arg_Spec_List, string)
+        if result:
+            # There is a match so check the number of args provided
+            # matches the number of args expected by the intrinsic.
+            function_name = str(result[0])
+            function_args = result[1]
+            # This if/else will not be needed once issue #170 has been
+            # addressed.
+            if (isinstance(function_args, Actual_Arg_Spec_List)):
+                nargs = len(function_args.items)
+            elif function_args is None:
+                nargs = 0
+            else:
+                nargs = 1
+
+            if function_name in Intrinsic_Name.specific_function_names.keys():
+                # If this is a specific function then use its generic
+                # name to test min and max number of arguments.
+                test_name = Intrinsic_Name.specific_function_names[
+                    function_name]
+            else:
+                test_name = function_name
+
+            min_nargs = Intrinsic_Name.generic_function_names[test_name]["min"]
+            max_nargs = Intrinsic_Name.generic_function_names[test_name]["max"]
+
+            if min_nargs == max_nargs and nargs != min_nargs:
+                raise InternalSyntaxError(
+                    "Intrinsic '{0}' expects {1} arg(s) but found {2}."
+                    "".format(function_name, min_nargs, nargs))
+            if min_nargs < max_nargs and (nargs < min_nargs or
+               nargs > max_nargs):
```

Comment:
pylint complains about the indentation here.

Reply:
Fixed. Sorry, I'm not sure how the pylint errors got through.